scripts/loadEmptyPrefsSets.js

```diff
@@ -0,0 +1,103 @@
+/*!
+Copyright 2018 Raising the Floor US
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+// This script generates unique pref sets and loads them into CouchDB.
+//
+// See this JIRA for more information: https://issues.gpii.net/browse/GPII-3381
+// The options to this script must be set as environment variables when calling it,
+// these are:
+//   * COUCHDB_URL: This neeeds to be in the form: "http://user:pass@host:port/dbname"
+//   * NEW_USERS: The amount of new empty preferences sets you want to create
+//
+
+"use strict";
+
+var process = require("process"),
+    fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    uuid = uuid || require("node-uuid");
+
+require("./shared/prefsSetsDbUtils.js");
+require("gpii-pouchdb");
+
+
+fluid.defaults("gpii.uuidLoader", {
+    gradeNames: ["gpii.pouch"],
+    dbOptions: {
+        //name: "http://localhost:5984/gpii"
+        name: process.env.COUCHDB_URL
+    },
+    invokers: {
+        addPrefsSet: {
+            funcName: "gpii.uuidLoader.addPrefsSet",
+            args: [ "{that}", "{arguments}.0", "{arguments}.1"]
+        },
+        dbPut: {
+            funcName: "gpii.uuidLoader.dbPut",
+            args: [ "{that}", "{arguments}.0"]
+        }
+    }
+});
+
+gpii.uuidLoader.dbPut = function (that, doc) {
+    var promise = fluid.promise();
+    // We only perform the put after checking that the uuid doesn't exist in DB
+    that.get(doc._id).then(function (/* record */) {
+        promise.reject({isError: true, message: "docId already exists in DB"});
+    }, function (err) {
+        if (err.message === "missing") {
+            that.put(doc).then(function (newDoc) {promise.resolve(newDoc);}, function (err) {promise.reject(err);});
+        } else {
+            promise.reject(err);
+        };
+    });
+    return promise;
+};
+
+gpii.uuidLoader.addPrefsSet = function (that, gpiiKey, prefSafe) {
+    var promise = fluid.promise();
+    var sequence = [
+        that.dbPut(gpiiKey),
+        that.dbPut(prefSafe)
+    ];
+
+    fluid.promise.sequence(sequence).then(function () {
+        promise.resolve(gpiiKey._id);
+    }, function (error) {
+        promise.reject(error);
+    });
+
+    return promise;
+};
+
+// The real action
+//
+function createNewUsers(uuidLoader) {
+    var data = gpii.prefsSetsDbUtils.generatePrefsSet(uuid.v4(), gpii.prefsSetsDbUtils.emptyPrefsSetBlock);
+    uuidLoader.addPrefsSet(data.key, data.prefsSafe).then(function (newUser) {
+        console.log("## New user created:", newUser);
```

Comment:
Given our discussion about error conditions, I think it might be helpful to include the value of `n` here, as in "User #n created".

If you're feeling fancy, save the original `n` so you can say things like "User 5 of 20 created". :)